src/Slack/SlackChannel.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Illuminate\Notifications\Slack;
+
+use Illuminate\Http\Client\Factory;
+use Illuminate\Http\Client\Response;
+use Illuminate\Notifications\Notification;
+use Illuminate\Support\Facades\Config;
+
+class SlackChannel
+{
+    /**
+     * The HTTP factory instance.
+     *
+     * @var \Illuminate\Http\Client\Factory
+     */
+    protected $http;
+
+    /**
+     * Create a new Slack channel instance.
+     *
+     * @return void
+     */
+    public function __construct(Factory $http)
+    {
+        $this->http = $http;
+    }
+
+    /**
+     * Send the given notification.
+     */
+    public function send(mixed $notifiable, Notification $notification): ?Response
+    {
+        $route = $this->determineRoute($notifiable, $notification);
+
+        $message = $notification->toSlack($notifiable);
+
+        $payload = $this->buildJsonPayload($message, $route);
+
+        if (! $route->token || ! $payload['channel']) {
+            return null;
+        }
+
+        return $this->http->asJson()
+            ->withToken($route->token)
+            ->post('https://slack.com/api/chat.postMessage', $payload);
```

Comment:
Yeah, that actually makes a lot of sense. Fixed. 👍 

It'll now throw on non-200's, as well as when the response itself gets checked, and if a Slack "non-ok" response is given, it'll throw an exception that includes the "error" property from the error. This can range from permission issues, to non-existent channels and even invalid authentication tokens.